ui/selectors/selectors.js

```diff
@@ -1757,6 +1762,25 @@ export function getCustomTokenAmount(state) {
   return state.appState.customTokenAmount;
 }
 
+export function getUpdatedAndSortedAccounts(state) {
+  const accounts = getMetaMaskAccountsOrdered(state);
+  const pinnedAccounts = getPinnedAccountsList(state);
```

Comment:
if we return a set here (instead of an array) we can reduce the lookup cost in the following line from O(n) to O(1) since `set.has(element)` is significantly more performant.

Reply:
https://www.tech-hour.com/javascript-performance-and-optimization